pycode/memilio-epidata/memilio/epidata/assessNPIEffects.py

```diff
@@ -0,0 +1,595 @@
+        #                make_plot=dd.defaultDict['make_plot'],
+        #                )
+
+    else:  # read formatted file
+        npis = pd.read_excel(
+            os.path.join(
+                directory, 'datensatzbeschreibung_massnahmen.xlsx'),
+            sheet_name=3, engine='openpyxl')
+        if fine_resolution > 0:
+            if fine_resolution == 1:
+                filename = 'germany_counties_npi_subcat_incgrouped'
+            else:
+                filename = 'germany_counties_npi_subcat'
+        else:
+            filename = 'germany_counties_npi_maincat'
+        df_npis = pd.read_csv(directory + filename + ".csv")
+    try:
+        df_npis = df_npis.drop('Unnamed: 0', axis=1)
+    except KeyError:
+        pass
+    # df_npis = mdfs.extract_subframe_based_on_dates(df_npis, date(2021,1,1), date(2021,6,1))
+    npis = pd.read_json(os.path.join(directory, 'npis.json'))
+    # get code levels (main/subcodes) and position of main codes
+    # code_level = [i.count('_') for i in npi_codes]
+    # main_code_pos = [i for i in range(len(code_level)) if code_level[i] == 1]
+
+    # check if any other integer than 0: not implemented or 1: implemented is
+    # used (maybe to specify the kind of implementation)
+
+    all_subcodes = [x for x in npis.NPI_code if len(x) <=8]
+    df_merged = df_npis.iloc[:, :2]
+    for subcode in all_subcodes:
+        # extract columns which have the subcode as part of the column
+        # name and sum over all these subcodes
+        df_merged[subcode] = df_npis.filter(
+            regex=subcode).sum(axis=1)
+    #npi_codes_considered = npis.NPI_code.values.tolist()
+    npi_codes_considered = all_subcodes
+    df_npis = df_merged.copy()
+
+    if len(np.where(df_npis[npi_codes_considered] > 1)[0]) > 0:
+
+        print("Info: Please ensure that NPI information is only boolean.")
+
+    else:
+        # sum over different NPIs and plot share of countires implementing
+        # these NPIs versus counties without corresponding actions
+        df_npis_aggregated = df_npis.groupby(
+            dd.EngEng['date']).agg(
+            {i: sum for i in npi_codes_considered}).copy()
+        npis_total_sum = df_npis_aggregated.sum()
+
+        npi_codes_empty = list(np.array(npi_codes_considered)[
+                               np.where(npis_total_sum == 0)[0]])
+
+        npi_unused_indices_all = []
+        npi_used_indices_all = []
+        npi_unused_indices = []
+        npi_used_indices = []
+        for i in range(len(npi_codes_considered)):
+            if npi_codes_considered[i] in npi_codes_empty:
+                npi_unused_indices.append(i)
+                npi_unused_indices_all.append(
+                    np.where(npis[dd.EngEng['npiCode']] == npi_codes_considered[i])[0][0])
+            else:
+                npi_used_indices.append(i)
+                npi_used_indices_all.append(
+                    np.where(npis[dd.EngEng['npiCode']] == npi_codes_considered[i])[0][0])
```

Comment:
npis works on all codes (M01a_010_0, M01a_010_1, ...) while npi_codes_considered is only on first level codes (i.e. M01a_010, M01a_020, ...)

Reply:
changed npis